provides.py

```diff
@@ -30,10 +30,11 @@ def departed(self):
         conv = self.conversation()
         conv.remove_state('{relation_name}.joined')
 
-    def set_ready(self, port, controllers):
+    def set_ready(self, port, api_path, controllers):
```

Comment:
I'll rework this a bit to use individual methods in a future PR.  kwargs might be easier, but I like the explicitness of separate methods.